src/chat/channel.py

```diff
@@ -241,9 +241,14 @@ def printLine(self, chname, text, scroll_forced=False, formatter=Formatters.FORM
         if player.clan is not None:
             displayName = "<b>[%s]</b>%s" % (player.clan, chname)
 
+        name_of_author_message=''
+        if hasattr(player, 'login'):
```

Comment:
There's no need to do it like this if player is a Player instance.

Reply:
Maybe I'm missing your point but what are you thinking then ? 

If I'm not wrong, either the user is an IRCPlayer (if only irc) or Player instance (if he can also play games & co).
If it's an ircplayer then its name is in .name, but for player the name is in .login. So that's why I did like this, otherwise I can't grab his username (or will crash since no attribute).

Reply:
Using hasattr is a worse idea than using isinstance. Either way, neither of these things should be necessary - we need to check whether the chatter is us, and I think the better way to do it is to check whether his corresponding player is the same player we're logged in as (or, if chatter.player is client.me.player and both are not None).

Reply:
Ok I did it as you said but tbh I'm not sure what you were thinking because I'm confused between all these classes and your "if" only works when he's not an irc user (if irc : login_user, a new variable I added, is None) so if a irc pings me I won't see the message in red ( maybe I'm wrong somewhere in the logic )